packages/terra-open-source-scripts/src/terra-cli/release/releaseRepoHandler.js

```diff
@@ -0,0 +1,39 @@
+const pacote = require('pacote');
+const path = require('path');
+const fs = require('fs-extra');
+const childProcess = require('child_process');
+const { promisify } = require('util');
+const Logger = require('@cerner/terra-cli/lib/utils/Logger');
+
+const setupGit = require('./setupGit');
+
+const logger = new Logger({ prefix: '[terra-open-source-scripts:release]' });
+
+const exec = promisify(childProcess.exec);
+
+const isPublished = async (name, version) => {
+  const pkgJson = await pacote.packument(name, {
+    registry: 'https://registry.npmjs.org/',
+  });
+  const publishedVersions = Object.keys(pkgJson.versions);
+  return publishedVersions.includes(version);
+};
+
+module.exports = async () => {
+  if (await isPublished()) {
+    logger.info('Nothing to publish');
+    return;
+  }
+  const packageFile = path.resolve(process.cwd(), 'package.json');
+
+  // Read package.json and pull out version
+  const { name, version } = await fs.readJSON(packageFile);
+
+  await setupGit(name, version);
+
+  const tag = `v${version}`;
+
+  await exec('npm publish');
```

Comment:
can we verify successful publish before tagging? or will the script blow up if this command fails?

Reply:
i believe it'll blow up if it fails, but i'll verify.

Reply:
I verified that an exception is thrown if this exec fails.